internal/config/config.go

```diff
@@ -0,0 +1,116 @@
+package config
+
+import (
+	"fmt"
+	"github.com/rs/zerolog"
+	"strings"
+	"sync"
+	"time"
+
+	clowder "github.com/redhatinsights/app-common-go/pkg/api/v1"
+	"github.com/spf13/viper"
+)
+
+type GlobalConfig struct {
+	Database struct {
+		Host        string
+		Port        uint16
+		Name        string
+		User        string
+		Password    string
+		MaxIdleTime time.Duration
+		MaxLifetime time.Duration
+		MaxOpenConn int
+		MaxIdleConn int
+		LogLevel    int
+	}
+	Logging struct {
+		Level  int
+		Stdout bool
+	}
+	Cloudwatch struct {
+		Enabled bool
+		Group   string
+		Stream  string
+	}
+	FeatureFlags struct {
+		Environment string
+	}
+	ExitOnPanic bool
+}
+
+var config GlobalConfig
+var configMutex sync.Mutex
+
+func init() {
+	var err error
+
+	viper.SetDefault("logging.level", 1)
+	viper.SetDefault("database.host", "localhost")
+	viper.SetDefault("database.port", 5432)
+	viper.SetDefault("database.loglevel", 1)
+	viper.SetDefault("cloudwatch.enabled", false)
+
+	if clowder.IsClowderEnabled() {
+		fmt.Println("Loading Clowder configuration")
+		cfg := clowder.LoadedConfig
+
+		viper.Set("database.host", cfg.Database.Hostname)
+		viper.Set("database.port", cfg.Database.Port)
+		viper.Set("database.user", cfg.Database.Username)
+		viper.Set("database.password", cfg.Database.Password)
+		viper.Set("database.name", cfg.Database.Name)
+	} else {
+		fmt.Println("Loading local configuration")
+
+		// read missing default environmental values from .env.default
+		viper.AddConfigPath("./configs")
+		viper.SetConfigName("defaults")
+		err = viper.ReadInConfig()
+		if err != nil {
+			panic(err)
+		}
+		viper.SetConfigName("local")
+		err = viper.MergeInConfig()
+		if err != nil {
+			fmt.Println("Could not read local.yaml", err)
+		}
+	}
+
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+
+	err = viper.Unmarshal(&config)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func Get() *GlobalConfig {
+	configMutex.Lock()
+	defer configMutex.Unlock()
+
+	return &config
+}
```

Comment:
Actually would you mind simply removing the `Get()` alltogether? There is no need for it since it is no longer lazy-initialized. Also remove the Global struct and simply expose the substructs, perhaps move ExitOnPanic to Logging struct. We will need to use configuration a lot and this is much nicer to write:

    config.Database.Password

Reply:
I've tried my best, do you think this is good enough implementation? :)

Reply:
It's great except the naming and unused `config` variable, see above.